subprojects/core/src/main/java/org/gradle/internal/metaobject/BeanDynamicObject.java

```diff
@@ -54,6 +54,14 @@
  */
 public class BeanDynamicObject extends AbstractDynamicObject {
     private static final Method META_PROP_METHOD;
+    private static final ThreadLocal<Object[]> META_PROP_ARGUMENT_ARRAY = new ThreadLocal<Object[]>() {
```

Comment:
:x: This should be reverted: it will create a similar memory leak as the one fixed in `1d78cf1`. I also doubt it has an impact in real world: profilers will prevent the VM from eliminating the array creation.

Reply:
JFR takes escape analysis into account. So this memory usage is real.

Also, this is not the same as the other issue, because this is a static field, not an instance field.

Reply:
Right, but this is not strictly escape analysis, it's about avoiding the creation of an array, which, to my knowledge, the JVM is capable of doing (typically because of reflective calls). I still doubt this is necessary. Do we have evidence it makes a difference?

Reply:
Since JFR instruments new TLABs, I don't think anything is being optimized. Otherwise it wouldn't be allocated. It's about 100MB of garbage when creating 500K tasks in a loop. So for a typical Gradle project you won't feel a thing. It only gets relevant when you try pushing our current scaling limits.

I'm okay with deferring this until we make our test projects bigger so I can demonstrate it more clearly.